java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -1994,7 +1995,18 @@ private static byte[] commonReadByteArrays(InStream stream, IntegerReader length
           totalLength = (int) (batchSize * scratchlcv.vector[0]);
         }
       }
-
+      if (totalLength < 0) {
+        StringBuilder sb = new StringBuilder("totalLength:" + totalLength
+                + " is a negative number.");
+        if (batchSize > 1) {
```

Comment:
This PR seems to be mixing three things. 

First, when `totalLength < 0` happens, do we need to throw a new exception from ORC side?

Second, in the above case, when `batchSize == 1` additionally, this PR is adding a new exception here. Logically, the following.
```java
if (totalLength < 0 && batchSize == 1) {
    throw new IOException("totalLength:" + totalLength + " is a negative number."
}
```
- If this is a meaningful error, can we proceed that in a separate PR because we may want to backport that.

Reply:
It makes sense to add this exception.
Because when the user encounters the `NegativeArraySizeException`, user does not know what to do. The user can only read the source code of the orc to know that the batch size can be adjusted.

I can remove it in this PR first and then raise another jira to add this exception.

I encountered this problem when reading orc datasource in Spark 3.2, because the orc structure is complex, so the default is to use `OrcMapreduceRecordReader` to read.
Then I use config
```
spark.sql.orc.enableNestedColumnVectorizedReader=true
spark.sql.orc.columnarReaderBatchSize=1
```
Temporarily solved the problem.

Reply:
Yes, I agree with you that the directional error message will be helpful.